rubix-core/src/main/java/com/qubole/rubix/core/CachingInputStream.java

```diff
@@ -245,7 +246,10 @@ private int readInternal(byte[] buffer, int offset, int length)
     final List<ReadRequestChain> readRequestChains = setupReadRequestChains(buffer,
         offset,
         endBlock,
-        length);
+        length,
+        nextReadPosition,
```

Comment:
Instead of taking `nextReadPosition` and `nextReadBlock`, can we just make this two class variables non private. We are passing the same value that is set in CachingInputStream

Reply:
We cant do this because setupReadRequestChains on same InputStream in dummy mode can be called in parallel by multiple threads because we offload it to executor in every read.  Reusing the class members causes issues because threads now step over each other, hence dummy InputStream stores the required variables locally and pass it to this function.